AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -105,26 +105,20 @@ + (void)setLogUrl:(NSString *)logUrl {
 }
 
 + (void)setEnabled:(BOOL)isEnabled {
-  @synchronized([MSAppCenter sharedInstance]) {
-    if ([[MSAppCenter sharedInstance] canBeUsed]) {
-      [[MSAppCenter sharedInstance] setEnabled:isEnabled];
-    }
+  if ([[MSAppCenter sharedInstance] canBeUsed]) {
```

Comment:
We probably still need some synchronization between `canBeUsed` and `setEnabled`. Either we leave it here or we move the `canBeUsed` check in the instance method in the synchronized.